# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.cpp
# Commit: f82a44c95497
# Full Hash: f82a44c95497c4821a758ba031078945a188bc21
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-16 15:49:59
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r af6b36a497e8 -r f82a44c95497 js/src/vm/JitActivation.cpp
--- a/js/src/vm/JitActivation.cpp	Tue Jun 16 08:54:41 2020 +0000
+++ b/js/src/vm/JitActivation.cpp	Tue Jun 16 09:26:20 2020 +0000
@@ -231,9 +231,9 @@
   MOZ_ASSERT(unwound == (trap == wasm::Trap::IndirectCallBadSig));
 
   void* pc = unwindState.pc;
-  wasm::Frame* fp = unwindState.fp;
+  const wasm::Frame* fp = wasm::Frame::fromUntaggedWasmExitFP(unwindState.fp);
 
-  const wasm::Code& code = fp->tls->instance->code();
+  const wasm::Code& code = fp->instance()->code();
   MOZ_RELEASE_ASSERT(&code == wasm::LookupCode(pc));
 
   // If the frame was unwound, the bytecodeOffset must be recovered from the